designdocs/provider_contexts.md

```diff
@@ -0,0 +1,323 @@
+
+Finally, this structure does not prevent the setup of a 1:1 relationship between Application and Provider.
+If a company wishes, they can create more than one Provider for the purposes of separation.
+This may be desirable if there is a subsidiary company or other complex structures.
+
+#### User Trust in Application Presentation
+
+Users need to know what application they are logging into and trust that the displayed information is not a phishing attack.
+Users have a reasonable expectation that the chain and wallet provide a level of protection against phishing attacks from Providers and Applications.
+
+#### Easy Wallet Integration of Data
+
+Wallets displaying login request information need to be able to easily obtain and verify the information they want to show the user.
+
+## Proposal
+
+Frequency should provide a safeguarded system of setting and updating Provider and Provider Application Contexts for wallets to display to users.
+
+- [Storage and Structure of the Data](#data)
+- [Mainnet Approval Flow](#governance)
+- [Example of Wallet Usage via SIWF](#siwf)
+- [Provider Dashboard Steps](#dashboard)
+
+### **Storage and Structure of the Data** <a id='data'></a>
+
+Currently, the `ProviderRegistryEntry` has only a `name` property, and no way to update said name.
+
+Applications have two different pieces of default and (potentially) internationalized data:
+
+- name
+- logo/image
+
+Limits:
+
+1. Provide a space to internationalize up to 150 localizations (Windows and macOS each have < 150).
+2. Image MUST BE encoded (`png`)[https://www.w3.org/TR/png-3/] that is `250x100` (support for future sizes and light/dark should be considered).
+3. Localization keys should follow BCP 47 format like en-US, fr-FR.
+
+The data structure must support both of these and internationalization.
+
+Proposed are the following changes:
+
+1. Update `ProviderRegistryEntry`.
+
+    Each registry entry supports:
+    - A default name and logo.
+    - Localized names and logos (up to 150 locales).
+    - PNG logo support (250x100, up to 128 KiB).
+    - BCP 47-compliant language codes (e.g., en-US, fr-FR).
+2. Updated `ProviderRegistryEntry` struct have the following properties:
+
+    ```rust
+        pub struct ProviderRegistryEntry<T::Config> {
+            pub default_name: BoundedVec<u8, T::MaxProviderNameSize>,
+            pub localized_names: BTreeMap<Vec<T::MaxLanguageCodeSize>, BoundedVec<u8, T::MaxProviderNameSize>>,
+            pub default_logo_250_100_png_bytes: BoundedVec<u8, T::MaxProviderLogo250X100Size>,
+            pub localized_logo_250_100_png_bytes: BTreeMap<Vec<T::MaxLanguageCodeSize>, BoundedVec<u8, T::MaxProviderLogo250X100Size>>,
+        }
+    ```
+
+3. New `ProviderToApplicationRegistryEntry` storage be initialized:
+
+    ```rust
+    // Alias for clarity
+    type ApplicationIdentifier<T: Config> = BoundedVec<u8, T::MaxProviderNameSize>
+
```

Comment:
@wilwade do we need a special application identifiers? These are per provider basis so we can just use incremental number 0-Nth app? Or is there a utility tied to it

Reply:
I was thinking just incremental ids. Names could change (Twitter -> 𝕏).

The application name doesn't have to be unique from a chain perspective, but I cannot see to many reasons for duplicates... Or well if you are trying out different branding you might create a second app context with the new logo and then you can roll it out slower.